### PR TITLE
Fix pmap named axes

### LINKED ISSUE #111: Unbound axis name in filter_pmap + lax.pmean
I tried to parallelize my training code using the standard jax idiom of `pmap`ing the entire `update` function and performing `pmean` reduction on loss and gradients inside `update`. Unfortunately, I get an error about the mapped axis name being unbound. The same code works with `filter_vmap`, which I think is supposed to be API-equivalent, so I believe I am experiencing a bug rather than mere misuse. The reproducing code can be found [here](https://gist.github.com/jatentaki/779c949b1ff4e9ea1828fd7098fb0792).

### PR DESCRIPTION
**Pull request by jatentaki**
Fixes #111

### CODE REVIEW

**Comment by patrick-kidger** [comment]
LGTM! Can you bump the version number too? I'll do a new release.

**Comment by jatentaki** [comment]
There you go

**Comment by patrick-kidger** [comment]
Thanks for the contribution!
